File: iteration2/test/models/UserTest.java
package iteration2.test.models;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

import org.junit.Test;

import iteration2.src.models.Student;
import iteration2.src.models.User;

public class UserTest {

    @Test
    public void testCheckCredentials() {
        User user = new Student("123", "password", "John", "Doe", true, null, null, 0, null, null, null, null);
        assertTrue(user.checkCredentials("123", "password"));
        assertFalse(user.checkCredentials("123", "wrongpassword"));
    }

}

File: iteration3/src/test/UserTest.py
import unittest
from models.User import Student

class UserTest(unittest.TestCase):

    def testCheckCredentials(self):
        user = Student("123", "password", "John", "Doe", True, None, None, 0, None, None, None, None)
        self.assertTrue(user.check_credentials("123", "password"))
        self.assertFalse(user.check_credentials("123", "wrongpassword"))

if __name__ == '__main__':
    unittest.main()
